lib/routes/users/routes.js

```diff
@@ -59,5 +75,129 @@ module.exports = [
+            cors: true,
+            auth: false,
+            validate: {
+                payload: Joi.object({
+                    id: idAllowedSchema,
+                    firstName: stringSchema,
+                    lastName: stringSchema,
+                    mobileNo: stringSchema,
+                    email: emailSchema,
+                    gender: Joi.string().valid('male', 'female', 'other')
+                })
+            }
+        },
+        handler: async (request, h) => {
+            try {
+                await updateUser(request.payload);
+                return h
+                    .response({ message: 'User updated successfully.' })
+                    .code(200);
+            } catch (error) {
+                return h.response(error.message).code(400);
+            }
+        }
+    },
+    {
+        method: 'POST',
+        path: '/find/cab',
+        options: {
+            description: 'find nearby cabs',
+            notes: 'POST users API',
+            tags: ['api', 'users'],
+            cors: true,
+            auth: false,
+            validate: {
+                payload: Joi.object({
+                    pickupLocation: Joi.array().items(Joi.number())
+                })
+            }
+        },
+        handler: async (request, h) => {
+            try {
+                let cab = transformDbArrayResponseToRawResponse(
+                    await findCab(request.payload.pickupLocation)
+                );
+                return h.response(cab).code(200);
+            } catch (error) {
+                return h.response(error.message).code(400);
+            }
+        }
+    },
+    {
+        method: 'POST',
+        path: '/attempt_to_book_cab',
+        options: {
+            description: 'attempt to book a cab',
+            notes: 'POST users API',
+            tags: ['api', 'users'],
+            cors: true,
+            auth: false,
+            validate: {
+                payload: Joi.object({
+                    userId: idAllowedSchema,
+                    pickupLocation: Joi.array().items(Joi.number()),
+                    dropLocation: Joi.array().items(Joi.number()),
+                    fare: Joi.number().required(),
+                    rideStatus: Joi.string().valid('pending')
+                })
+            }
+        },
+        handler: async (request, h) => {
+            try {
+                const { pickupLocation, dropLocation } = request.payload;
+                request.payload.pickupLocation = Sequelize.fn(
+                    'ST_GeomFromText',
+                    `POINT(${pickupLocation[0]} ${pickupLocation[1]})`
+                );
+                request.payload.dropLocation = Sequelize.fn(
+                    'ST_GeomFromText',
+                    `POINT(${dropLocation[0]} ${dropLocation[1]})`
+                );
+                let attemptToBook = convertDbResponseToRawResponse(
+                    await attemptToBookCab(request.payload)
+                );
```

Comment:
Let's move this logic to the Dao. A small exercise you can attempt within the dao is to actually do is using this convertDbResponse / ArrayResponse to accept a promise and resolve it internally. 

so you would call it `await converDbResponseToRawResponse(foo)`. Notice the lack of await in the param. The function will resolve the recieved param.